### PR TITLE
Feat: Expand bids count for auctions

### PR DESCRIPTION
**Pull request by zaxozhu**
This PR fixes selling more than 80 editions by having an external BidStateData account to store bids for big auctions.

### CODE REVIEW

File: js/packages/web/.env [hunk 1]
```diff
@@ -1,4 +1,4 @@
-REACT_APP_STORE_OWNER_ADDRESS_ADDRESS=
+REACT_APP_STORE_OWNER_ADDRESS_ADDRESS=Hu5rhimipWHGUZRQZtYZkQ6PLD8rTCiFtsPzXQ1cmxvL
```
Comment (kurpav) [inline js/packages/web/.env:1]: probably should be empty

File: js/packages/web/src/actions/createAuctionManager.ts [hunk 2]
```diff
@@ -138,11 +141,33 @@ export async function createAuctionManager(
     fractionTreasury,
   } = await createVault(connection, wallet, priceMint, externalPriceAccount);
 
+  // AuctionData account can handle N <= 80 max
+  // If there are more bids, move BidState to BidStateData account
+  let bidStateDataInstructions: TransactionInstruction[] = [];
+  let bidStateDataSigners: Keypair[] = [];
+  let bidStateDataAccount;
+
+  if (auctionSettings.winners.usize.toNumber() > 80) {
```
Comment (kurpav) [inline js/packages/web/src/actions/createAuctionManager.ts:150]: what is 80 ?

File: js/packages/common/src/utils/ids.ts [hunk 3]
```diff
@@ -66,16 +66,16 @@ export const MEMO_ID = new PublicKey(
 );
 
 export const METADATA_PROGRAM_ID =
-  'metaqbxxUerdq28cj1RbAWkYQm3ybzjb6a8bt518x1s' as StringPublicKey;
+  '97N7MM9VDDeJmk9CH9wKCQG3dn1wgpGaedYbq5Vey17c' as StringPublicKey;
```
Comment (VadimGrozinok) [inline js/packages/common/src/utils/ids.ts:69]: Change back to Metaplex token-metadata program key

File: js/packages/common/src/utils/ids.ts [hunk 4]
```diff
@@ -66,16 +66,16 @@ export const MEMO_ID = new PublicKey(
...
 
 export const VAULT_ID =
   'vau1zxA2LbssAUEF7Gpw91zMM1LvXrvpzJtmZ58rPsn' as StringPublicKey;
 
 export const AUCTION_ID =
-  'auctxRXPeJoc4817jDhf4HbjnhEcr1cCXenosMhK5R8' as StringPublicKey;
+  '9qyd7zhgK8nTGF2mnjLzzo3Z4NRLH35o4Q69PnaL3tJ' as StringPublicKey;
```
Comment (VadimGrozinok) [inline js/packages/common/src/utils/ids.ts:75]: The same as above

File: js/packages/common/src/utils/ids.ts [hunk 5]
```diff
@@ -66,16 +66,16 @@ export const MEMO_ID = new PublicKey(
...
 
 export const METAPLEX_ID =
-  'p1exdMJcjVao65QdewkaZRUnU6VPSXhus9n2GzWfh98' as StringPublicKey;
+  '8AqKV7JS4D8s9t5JMJS4cHGcnQT8ZjV994iWb8xWCjyj' as StringPublicKey;
```
Comment (VadimGrozinok) [inline js/packages/common/src/utils/ids.ts:78]: The same as above

File: rust/auction/program/src/processor.rs [hunk 6]
```diff
@@ -350,33 +360,37 @@ impl AuctionData {
         };
     }
 
-    pub fn is_winner(&self, key: &Pubkey) -> Option<usize> {
-        let minimum = match self.price_floor {
+    /// TODO: Provide implementation for methods below according to new bids count refactoring.
```
Comment (VadimGrozinok) [inline rust/auction/program/src/processor.rs:363]: Is this comment relevant or can be removed ?

**Review by VadimGrozinok — CHANGES_REQUESTED** [review]

File: rust/auction/program/src/processor.rs (hunk 6 above)
Comment (bezrazli4n0) [reply rust/auction/program/src/processor.rs:363]: > Is this comment relevant or can be removed ?

Can be removed

Comment (VadimGrozinok) [reply rust/auction/program/src/processor.rs:363]: Then remove please 🙂

**Pull request closed without merging**